mobile/examples/basic_usage/ios/OrtBasicUsage/OrtBasicUsage/SwiftOrtBasicUsage.swift

```diff
@@ -0,0 +1,109 @@
+  case error(_ message: String)
+}
+
+// Some helper functions to copy values out of/into a Data instance.
+
+private func dataCopiedFromArray<T>(_ array: [T]) -> Data {
+  return array.withUnsafeBufferPointer { buffer -> Data in
+    return Data(buffer: buffer)
+  }
+}
+
+private func arrayCopiedFromData<T>(_ data: Data) -> [T]? {
+  guard data.count % MemoryLayout<T>.stride == 0 else { return nil }
+  return data.withUnsafeBytes { bytes -> [T] in
+    return Array(bytes.bindMemory(to: T.self))
+  }
+}
+
+/// Adds `a` and `b` using ONNX Runtime.
+func SwiftOrtAdd(_ a: Float, _ b: Float) throws -> Float {
+  // We will run a simple model which adds two floats.
+  // The inputs are named `A` and `B` and the output is named `C` (A + B = C).
+  // All inputs and outputs are float tensors with shape [1].
+  guard
+    let modelPath = Bundle.main.path(
+      forResource: "single_add.basic",
+      ofType: "ort")
+  else {
+    throw SwiftOrtBasicUsageError.error("failed to get model path")
+  }
+
+  // First, we create the ORT environment.
+  // The environment is required in order to create an ORT session.
+  // ORTLoggingLevel.warning should show us only important messages.
+  let env = try ORTEnv(loggingLevel: ORTLoggingLevel.warning)
+
+  // Next, we will create some ORT values for our input tensors. We have two
+  // floats, `a` and `b`.
+  let valueDataType = ORTTensorElementDataType.float
+  let valueShape: [NSNumber] = [1]
+
+  // `aData` will hold the memory of the input ORT value.
+  // We set it to contain a copy of our input float, `a`.
+  let aData = dataCopiedFromArray([a])
+  // This will create a value with a tensor with the a copy of `a`'s data, of
+  // type float, and with shape [1].
+  let aInputValue = try ORTValue(
+    tensorData: NSMutableData(data: aData),
+    elementType: valueDataType,
+    shape: valueShape)
+
+  // And we do the same for `b`.
+  let bData = dataCopiedFromArray([b])
+  let bInputValue = try ORTValue(
+    tensorData: NSMutableData(data: bData),
+    elementType: valueDataType,
+    shape: valueShape)
+
+  // Now, we will create an ORT session to run our model.
+  // One can configure session options with a session options object
+  // (ORTSessionOptions).
+  // We use the default options with sessionOptions: nil.
+  let session = try ORTSession(env: env, modelPath: modelPath, sessionOptions: nil)
+
+  // With a session and input values, we have what we need to run the model.
+  // We provide a dictionary mapping from input name to value and a set of
+  // output names.
+  // This run method will run the model, allocating the output(s), and return
+  // them in a dictionary mapping from output name to value.
+  // As with session creation, it is possible to configure run options with a
+  // run options object (ORTRunOptions).
+  // We use the default options with runOptions: nil.
+  let outputs = try session.run(
+    withInputs: ["A": aInputValue, "B": bInputValue],
```

Comment:
Can we by any chance also show how to use the api to get the inputs of the graph too here? (I know for this case it's just simply "A" and "B" here) as you added the get inputs/outputs methods to the api.

Reply:
I added it (just in Swift for now)